R/subscript-tables.R

```diff
@@ -334,54 +408,24 @@ getQTestInfoIndexForVectorOutput <- function(evaluated.args, dimnames.x, perm)
     return(df.idx)
 }
 
-addArrayIndicesIfMissing <- function(q.test.info, y, dim.names)
+addArrayIndicesIfMissing <- function(q.test.info, y, dim.names, perm)
 {
     QTABLE.DIM.NAMES.ALLOWED <- c("Row", "Column", "Inner Row", "Outer Column",
                                   "Outer Row", "Inner Column")
     col.idx <- colnames(q.test.info) %in% QTABLE.DIM.NAMES.ALLOWED
     indices.already.present <- any(col.idx)
     if (indices.already.present)
         return(q.test.info)
-    arr.idx <- expand.grid(dim.names)
-
-    col.ord <- match(colnames(arr.idx), QTABLE.DIM.NAMES.ALLOWED)
-    if (all(!is.na(col.ord)))
-        arr.idx <- arr.idx[, QTABLE.DIM.NAMES.ALLOWED[sort(col.ord)],
-                           drop = FALSE]
+    arr.idx <- expand.grid(dim.names[perm])
+    if (NCOL(arr.idx) > 1)
+        arr.idx <- arr.idx[, names(dim.names)]
+    # col.ord <- match(colnames(arr.idx), QTABLE.DIM.NAMES.ALLOWED)
+    # if (all(!is.na(col.ord)))
+    #     arr.idx <- arr.idx[, QTABLE.DIM.NAMES.ALLOWED[sort(col.ord)],
+    #                        drop = FALSE]
```

Comment:
Recommend deleting old code if it is here as a comment. Can be confusing when revisiting the code later.